akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/RaffleExample.java

```diff
@@ -0,0 +1,195 @@
+ *
+ * but in Java with a mutable domain object (state) that is independent of Akka
+ */
+public class RaffleExample {
+
+  public static class RaffleId {}
+
+  interface Event {}
+  public static final class RaffleStarted implements Event { }
+  public static final class ParticipantAdded implements Event {
+    public final String name;
+    public ParticipantAdded(String name) {
+      this.name = name;
+    }
+  }
+  public static final class ParticipantRemoved implements Event {
+    public final String name;
+    public ParticipantRemoved(String name) {
+      this.name = name;
+    }
+  }
+  public static final class RaffleCompleted implements Event {}
+
+  interface Command {}
+  public static final class StartRaffle implements Command { }
+
+  public static final class AddParticipant implements Command {
+    public final String name;
+    public AddParticipant(String name) {
+      this.name = name;
+    }
+  }
+
+  public static final class RemoveParticipant implements Command {
+    public final String name;
+    public RemoveParticipant(String name) {
+      this.name = name;
+    }
+  }
+  public static final class CompleteRaffle implements Command { }
+
+  // TODO replies are not implemented in this example yet
+
+  private enum RaffleState {
+    NOT_STARTED,
+    RUNNING,
+    COMPLETED
+  }
+
+  // mutable domain model impl, completely free from Akka stuff
+  // requires https://github.com/akka/akka/issues/25740
+  static class Raffle {
+    private RaffleState state = RaffleState.NOT_STARTED;
+    private boolean finished = false;
+    private List<String> participants = new ArrayList<>();
+    private Optional<String> winner = Optional.empty();
+
+    RaffleState getState() {
+      return state;
+    }
+
+    boolean containsParticipant(String name) {
+      return participants.contains(name);
+    }
+
+    void start() {
+      state = RaffleState.RUNNING;
+    }
+
+    void addParticipant(String name) {
+      participants.add(name);
+    }
+
+    void removeParticipant(String name) {
+      participants.remove(name);
+    }
+
+    void complete() {
+      state = RaffleState.COMPLETED;
+      int winnderIndex = new Random().nextInt(participants.size() - 1);
+      winner = Optional.of(participants.get(winnderIndex));
+    }
+
+  };
+
+
+  public static Behavior<Command> behavior(String raffleId) {
+    return new PersistentBehavior<Command, Event, Raffle>(new PersistenceId("Raffle|" + raffleId)) {
+
+      @Override
+      public Raffle emptyState() {
+        return new Raffle();
+      }
+
+      @Override
+      public CommandHandler<Command, Event, Raffle> commandHandler() {
+        CommandHandlerBuilder<Command, Event, Raffle, Raffle> notStartedHandler =
+            commandHandlerBuilder(raffle -> raffle.getState() == RaffleState.NOT_STARTED)
+                .matchCommand(StartRaffle.class, (raffle, cmd) -> Effect().persist(new RaffleStarted()));
```

Comment:
Would it make sense to have convenience builder methods when using `null` state?

```java
public Raffle emptyState() { return null; }

CommandHandlerBuilder<Command, Event, Raffle, Raffle> notStartedHandler =
  
  commandHandlerBuilderNull()
    .matchCommand(StartRaffle.class, (cmd) -> Effect().persist(new RaffleStarted()));
...
}
```

Reply:
Btw, the name `commandHandlerBuilderNull` is horrible. I just want to drop the idea.

Reply:
Hmm, just saw that there a variant that accepts a Predicate and that we can write:

`commandHandlerBuilder(Objects::isNull)`

Reply:
there will be a match command with Function<Command> as alternative to Function<State, Command>, I think I added that in some other PR